### PR TITLE
Binyamin/initial release finishing touches

### PR DESCRIPTION
**Pull request by binyamin555**
Update the [README](https://github.com/xpodev/extype/blob/9b077d0637c748775ced381be1da662acd6e0f25/README.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -134,4 +137,102 @@ hatch run +py=39 dist:test
...
+Returns true if any of the elements, mapped through the given `fn`, is `True`.
+
+```py
+list.map(self: List[T], fn: Callable[[T], U]) -> List[U]
+```
+Returns a new list whose elements are the result of applying the given function on each element in the original list.
+
+```py
+list.reduce(self: List[T], fn: Callable[[T, T], T]) -> T
+```
```
Comment (elazarcoh) [inline README.md:164]: reduce function is incorrect, it should be:
```py
list.reduce(self: List[T], fn: Callable[[A, B], B]) -> T
```
Or A,B flipped, I actually don't remember. Only relevant with initial, actually.

File: README.md [hunk 2]
```diff
@@ -26,23 +26,26 @@ import extype
 ```
 
 Then, you can use the built-in extensions for the builtin types. In order to apply these extensions, 
-call the `extend_builtin_types` function:
+import the `extend_all` module from the `extype.builtin_extensions` package:
 ```py
-extype.extend_builtin_types()
+from extype.builtin_extensions import extend_all
 ```
```
[the remark this answers is not in the document]
Comment (elazarcoh) [reply README.md:33]: Okay, need to add an example of importing and extending some types only, and also list the avaliable modules.

[the remark this answers is not in the document]
Comment (binyamin555) [reply README.md:58]: I manually added this.

File: README.md (hunk 1 above)
Comment (binyamin555) [reply README.md:189]: You mean this?
```py
list.reduce(self: List[T], fn: Callable[[U, T], U], initial: U) -> U
```

Comment (elazarcoh) [reply README.md:190]: Yeah...

Comment (binyamin555) [reply README.md:190]: ok, resolved then

**Comment by binyamin555** [comment]
After merging this, I'll create a v1.0.0 release

**Review by elazarcoh — APPROVED** [review]

**Pull request merged**